### ISSUE TITLE
Cannot install tf-pugin cuda110 0.25 /w TF nightly

### ISSUE THREAD

**Issue by ben0it8** [NVIDIA/DALI#2256]
Hi,

I'm trying to upgrade my docker image to CUDA 11, for which TF 2.4 nightly is needed.
But, unfortunately, building the wheel for `nvidia-dali-tf-plugin-cuda11==0.25.0` fails because of the (known and noted) compiler mismatch issue.

Here's my environment info:

```
 Environment:
   ---------------------------------------------------------------------------------------------------------
   Platform system:                      Linux
   Platform machine:                     x86_64
   DALI lib path:                        /srv/venv/lib/python3.7/site-packages/nvidia/dali
   TF path:                              /srv/venv/lib/python3.7/site-packages/tensorflow
   DALI TF plugin destination directory: build/lib.linux-x86_64-3.7/nvidia/dali_tf_plugin
   Is Conda environment?                 No
   Using compiler:                       "g++", version 7.5
   TF version installed:                 2.4
   g++ version used to compile TF:       7.3
   Prebuilt plugins available for g++:   "4.8"
   Is g++-7.3 present in the system?     No
   Can install prebuilt plugin?          No
   Can compile with default compiler?    Yes
   Can compile with alt compiler?        No

```

Do you have any fix in mind to be able to install it with TF2.4? Note, that CUDA11 is **only** supported TF2.4+.

Cheers,
Oliver

**Comment by JanuszL** [comment]
Hi,
What error you exactly get? `Can compile with default compiler?    Yes` message says it should work.
Maybe this is the issue fixed by https://github.com/NVIDIA/DALI/pull/2236. It will be available in 0.25 or nightly build is you want to test it now.

**Comment by jantonguirao** [comment]
As Janusz pointed out, the environment info says that it can compile. After the environment information you shared, there should be the compiler command and probably a compiler error message. Can you share that with us? Then we'd know if that's the problem fixed by #2236

**Comment by ben0it8** [comment]
I've tested with nightly and it fixes my issue. It was actually the change in in tf.data.Dataset API. Thank you guys!
